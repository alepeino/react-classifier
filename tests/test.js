import React from 'react'
import renderer from 'react-test-renderer'
import C from '../src'

describe('plain structure', () => {
  describe('string classes', () => {
    test('root node by tag name', () => {
      const Comp = C(<div><p></p></div>, {
        'div': 'x'
      })
      const tree = renderer.create(Comp).toJSON()

      expect(tree.props.className).toEqual('x')
    })

    test('nested node by tag name', () => {
      const Comp = C(<div><p></p></div>, {
        'p': 'x'
      })
      const tree = renderer.create(Comp).toJSON()

      expect(tree.children[0].props.className).toEqual('x')
    })

    test('root node by root pseudo selector', () => {
      const Comp = C(<div><p></p></div>, {
        ':root': 'root',
        'p': 'nested'
      })
      const tree = renderer.create(Comp).toJSON()

      expect(tree.props.className).toEqual('root')
      expect(tree.children[0].props.className).toEqual('nested')
    })
  })
})
