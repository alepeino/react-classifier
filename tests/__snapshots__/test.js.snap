// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complex structure deeply nested 1`] = `
<div
  className="max-w-sm rounded overflow-hidden shadow-lg"
>
  <img
    alt="Sunset in the mountains"
    className="w-full"
    src="https://tailwindcss.com/img/card-top.jpg"
  />
  <div
    className="px-6 py-4"
  >
    <div
      className="font-bold text-xl mb-2"
    >
      The Coldest Sunset
    </div>
    <p
      className="text-grey-darker text-base"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptatibus quia, nulla! Maiores et perferendis eaque, exercitationem praesentium nihil.
    </p>
  </div>
  <div
    className="px-6 py-4"
  >
    <span
      className="inline-block bg-grey-lighter rounded-full px-3 py-1 text-sm font-semibold text-grey-darker mr-2"
    >
      #photography
    </span>
    <span
      className="inline-block bg-grey-lighter rounded-full px-3 py-1 text-sm font-semibold text-grey-darker mr-2"
    >
      #travel
    </span>
    <span
      className="inline-block bg-grey-lighter rounded-full px-3 py-1 text-sm font-semibold text-grey-darker"
    >
      #winter
    </span>
  </div>
</div>
`;
